### PR TITLE
Add interfaces and testing helpers for translation management systems

### PR DESCRIPTION
**Pull request by camertron**
Another branch in preparation for queuing. I'm in the process of abstracting a bunch of the commit status tracking from rosette-integration-smartling into rosette-core. I plan to rename rosette-integration-smartling to rosette-tms-smartling, and it will be responsible only for uploading phrases and downloading translations from Smartling. All the commit tracking logic and queuing will live in rosette-core.

@11mdlow @seunghyo @zvkemp

### CODE REVIEW

File: lib/rosette/core/translation_status.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      # Calculates a translated percentage for the given locale.
+      #
+      # @param [String] locale_code
+      # @param [Fixnum] precision The precision to use when rounding the
+      #   percentage.
+      # @return [Float] the translation percentage, rounded to +precision+
+      #   decimal places.
+      def percent_translated(locale_code, precision = 2)
+        pct = locale_counts[locale_code].to_f / phrase_count.to_f
+        pct > 1.0 ? 1.0 : pct.round(precision)
```
Comment (jdoconnor) [inline lib/rosette/core/translation_status.rb]: shoudl this be == 1.0?

**Comment by jdoconnor** [comment]
+1 with a small comment

File: lib/rosette/core/translation_status.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+      # Calculates a translated percentage for the given locale.
+      #
+      # @param [String] locale_code
+      # @param [Fixnum] precision The precision to use when rounding the
+      #   percentage.
+      # @return [Float] the translation percentage, rounded to +precision+
+      #   decimal places.
+      def percent_translated(locale_code, precision = 2)
+        pct = locale_counts[locale_code].to_f / phrase_count.to_f
```
Comment (11mdlow) [inline lib/rosette/core/translation_status.rb]: you probably only need to `#to_f` one of these

Comment (camertron) [inline lib/rosette/core/translation_status.rb]: Good call :)

File: lib/rosette/core/translation_status.rb (hunk 1 above)
Comment (camertron) [inline lib/rosette/core/translation_status.rb]: It certainly can be :) Trying to catch the case here where there are (for whatever reason) more translations than phrases, which would result in a > 100% translation percentage.

Comment (jdoconnor) [inline lib/rosette/core/translation_status.rb]:
```ruby
# return 100% at most, even though we can get more translations back than phrases
[1.0, pct.round(precision)].max
```

Comment (jdoconnor) [inline lib/rosette/core/translation_status.rb]: Max(1.0, Pct.round)

**Pull request merged**